kube-controllers/cmd/kube-controllers/main.go

```diff
@@ -121,14 +121,38 @@ func main() {
 	// Create the context.
 	ctx, cancel := context.WithCancel(context.Background())
 
+	// Create the status file. We will only update it if we have healthchecks enabled.
+	s := status.New(statusFile)
+
 	log.Info("Ensuring Calico datastore is initialized")
-	initCtx, cancelInit := context.WithTimeout(ctx, 10*time.Second)
-	defer cancelInit()
-	err = calicoClient.EnsureInitialized(initCtx, "", "k8s")
 	if err != nil {
 		log.WithError(err).Fatal("Failed to initialize Calico datastore")
 	}
 
+	s.SetReady("StartupCalicoDatastoreInitialization", false, "initialized to false")
+	initCtx, cancelInit := context.WithTimeout(ctx, 60*time.Second)
+	// Loop until context expires or calicoClient is initialized.
+	for {
+		select {
+		case <-initCtx.Done():
+			log.Fatal("Failed to initialize Calico datastore")
+			break
+		default:
```

Comment:
Very minor, but instead of a `default` you could do something like this:

```go
select {
  case <- ctx.Done():
    break
  case <- time.After(5*time.Second):
    // Try to initialize, break if successful. 
}
```

I find that to be slightly more idiomatic / easy to read.